vm-transfer/zvm-transfer.py

```diff
@@ -0,0 +1,455 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+"""
+A python script that clones a VM from the source system to the destination system
+"""
+
+__version__ = "0.1.0"
+
+import os
+import sys
+import logging
+import argparse
+import atexit
+import requests
+from pprint import pformat, pprint
+from munch import unmunchify
+
+sys.path.append('/opt/symphony-client')
+
+import symphony_client
+import config
+
+LOGS_DIR = "/var/log"
+STRATO_LOGS_DIR = LOGS_DIR + "/stratoscale"
+ZVM_CLONE_LOGS_DIR = STRATO_LOGS_DIR + "/zvm-clone"
+LOGGER_NAME = "zvm-clone"
+VPSA_VOLUME_TEMPLATE = 'neokarm_volume-{}'
+
+logger = logging.getLogger(LOGGER_NAME)
+
+arguments = None
+
+
+def migrate_vm(vm_id_or_name):
+    logger.info("vm_id: %s", vm_id_or_name)
+
+    src_symp_client = init_src_symp_client()
+    vm_list = src_symp_client.vms.list(detailed=True)
+    vm = get_vm_by_id(src_symp_client, vm_id_or_name, vm_list)
+    if not vm:
+        vm = get_vm_by_name(src_symp_client, vm_id_or_name)
+        if not vm:
+            sys.exit(1)
+    vm_id = vm.id
+    if vm.status == 'active':
+        logger.info("VM {} ({}) is active".format(vm.name, vm_id))
+        sys.exit(1)
+
+    dst_symp_client = init_dst_symp_client()
+    dest_vm = get_vm_by_name(dst_symp_client, vm.name)
+    if dest_vm:
+        logger.info("VM {} ({}) already exists in destination".format(dest_vm.name, dest_vm.id))
+        sys.exit(1)
+
+    networks = create_networking_map(src_symp_client, dst_symp_client, vm)
+    manageable_volumes, existing_volumes = check_volumes_in_dest(vm, dst_symp_client)
+
+    create_new_vm(vm, networks, manageable_volumes, existing_volumes, dst_symp_client)
+
+    logger.info("VM %s cloning is complete", vm.name)
+
+
+def get_vm_by_id(client, vm_id, vm_list=None):
+    if vm_list is None:
+        vm_list = client.vms.list(detailed=True)
+    vm = [vm for vm in vm_list if vm.id == vm_id]
+    if not vm:
+        logger.info("VM with id: %s does not exists", vm_id)
+        return None
+    return vm[0]
+
+
+def get_vm_by_name(client, vm_name, vm_list=None):
+    if vm_list is None:
+        vm_list = client.vms.list(detailed=True)
+    vm = [vm for vm in vm_list if vm.name == vm_name]
+    if not vm:
+        logger.info("VM with name: %s does not exists", vm_name)
+        return None
+    return vm[0]
```

Comment:
```suggestion
    return filtered_vm_list[0]
```

Reply:
Also in this method, need to test if len(filtered_vm_list) > 1 and raise exception for this.

Reply:
:+1: